### ISSUE TITLE
LESS: normalize icon mixins

### ISSUE THREAD

**Issue by ianmcburnie** [eBay/skin#276]
We have duplication of code across DS4 and DS6 icon mixins:

ds4/mixins.less
```less
.icon-play(@width: @ds4-icon-play-width; @height: @ds4-icon-play-height) {
    .background-icon-image(@width; @height; @ds4-icon-play-base64);
}
```
ds6/mixins.less
```less
.icon-play(@width: @ds6-icon-play-width; @height: @ds6-icon-play-height) {
    .background-icon-image(@width; @height; @ds6-icon-play-base64);
}
```

Let's see if we can normalize each ds4 and ds6 mixin into a common mixin without any ds specific prefixes (similar to how we've done the refactor in #615)

For example, we could consolidate too:

base/mixins.less
```less
.icon-play(@width: @icon-play-width; @height: @icon-play-height) {
    .background-icon-image(@width; @height; @icon-play-base64);
}
```

Where these variables are passed in from a ds4 or ds6 file.

We can only really do this for mixins and variables that exist in both ds4 and ds6. Those that don't will still need their prefix for now.

**Comment by ianmcburnie** [comment]
@agliga Actually, hold off on this one for now, I want to do a couple of experiments...

**Comment by ianmcburnie** [comment]
@seangates So I've been mulling over the idea of replacing all of these specific mixins with just a single generic mixin that takes a string parameter.

For example, instead of calling `.icon-play();`, calling `.background-icon-mixin('play');` instead.

Pros: less code to maintain
Cons: chance of invalid string

Thoughts? Is there some other good reason to keep the specific mixins that I'm missing/forgetting?

**Comment by seangates** [comment]
Well, if all icons behave exactly the same way, then we should use the mixin. If each needed to be custom (not likely going forward) then we should leave it as-is.

We could cut down on a bunch of LESS code, that's for sure. The string-in-a-LESS-variable-name trick is quite ingenious and can save us from a bunch of code.

**Comment by agliga** [comment]
We can actually add some code to guard against the con in that case, if we are on less 3.5 at least. 
I played around with this and it fails if you enter the wrong word. We can substitute this for the background image string or something (I used width for testing)
If you enter anything other than 'play' or 'plus', it will error. This can help us basically not worry about the con with a bit of extra code. 
```
@icons: {
  @play: 1px;
  @plus: 1px;
}

.mixin-icon(@str) {
  width: @icons[@@str];
}

.test {
  .mixin-icon('play');
}
```
